capstone-backend/app.js

```diff
@@ -137,56 +138,340 @@ app.post ('/profile/interests', async (req, res) => {
+      res.send(matchData)
+    }
+    else {
+      res.send({message : "no user found!" , typeStatus: "danger"})
+    }
+  }
+  catch (error){
+    res.send({message: "cannot get match data", typeStatus: "danger"})
+  }
+})
+
+async function updateMatch(params, currentUser){
+  const Match = Parse.Object.extend("Match");
+  const matchQuery = new Parse.Query(Match);
+  matchQuery.equalTo("user_1", currentUser.id);
+  matchQuery.equalTo("user_2", params.matchId);
+  let matchResults = await matchQuery.first();
+
+  matchResults.set("liked", params.liked);
+  await matchResults.save();
+  const privateInfo = new Parse.Query(Match);
+  privateInfo.equalTo("user_2", currentUser.id);
+  privateInfo.equalTo("user_1", params.matchId);
+  let privateInfoResults = await privateInfo.first()
+  if (privateInfoResults) {
+    privateInfoResults.set("display_private", params.liked);
+    await privateInfoResults.save()
+  }
+}
+
+async function createNewMatch(match, matchScore, user1, user2 ){
+  match.set("score", matchScore);
+  match.set("liked", false);
+  match.set("user_1", user1);
+  match.set("user_2", user2);
+  match.set("display_private", false);
+
+  await match.save()
+}
+
+function calculateScore(interest, category1, category2, weight1, weight2){
+  let user1Interest1 = [];
+  let user1Interest2 = [];
+  for (let i = 0; i < interest.user_1.length; i++){
+    user1Interest1 = user1Interest1.concat(interest.user_1[i].get(category1));
+    user1Interest2.push(interest.user_1[i].get(category2));
+  }
+  let user2Interest1 = [];
+  let user2Interest2 = [];
+  for (let i = 0; i < interest.user_2.length; i++){
+    user2Interest1 = user2Interest1.concat(interest.user_2[i].get(category1));
+    user2Interest2.push(interest.user_2[i].get(category2));
+  }
+  try {
+    let category1Score = compareRoles(user1Interest1, user2Interest1, weight1);
+    let category2Score = compareRoles(user1Interest2, user2Interest2, weight2);
+    let totalScore = category1Score + category2Score
+    return totalScore
+  }
+  catch (error){
+    console.log('error: ', error);
+    return 0;
+  }
+
+
+}
+
+function compareRoles(role1, role2, weight){
+  if (!role1 || !role2){
+    return 0;
+  }
+  let matches = 0;
+  for (let i = 0; i < role1.length; i++){
+    let match = false;
+    for (let j = 0; j < role2.length; j++){
+      if (role1[i] == role2[j]){
+        match = true;
+        break;
+      }
+    }
+    if (match){
+      matches++
+    }
+  }
+  let total = role1.length + role2.length;
+  if (total == 0){
+    return 0;
+  }
+  return ((matches/ total) * weight)
+}
+
+const skillWeight = 0.5;
+const skillNameWeight = 0.2;
+const rolesWeight = 0.3;
+
+function getScore(skills, roles){
+  const skillScore = calculateScore(skills, "category", "name", skillWeight, skillNameWeight);
+  const roleScore = compareRoles(roles.user_1, roles.user_2, rolesWeight);
+
+  return (skillScore + roleScore)
+}
+
+async function getMatches(currentUser){
```

Comment:
This function is actually writing to the DB after making calculations. A better name would be `createMatches` or `makeMatches`.
The name getMatches suggest a function reads information and returns it.